### PR TITLE
Added proof execution time to APRProof

### PR DESCRIPTION
**Pull request by lisandrasilva**
This PR records how long a proof takes to execute in the `APRProof` class so it can be accessed in Kontrol to display in the output message and in the `xml` report.

### CODE REVIEW

File: src/pyk/proof/reachability.py [hunk 1]
```diff
@@ -433,11 +445,14 @@ def write_proof_data(self) -> None:
         ensure_dir_path(self.proof_dir)
         ensure_dir_path(self.proof_subdir)
         proof_json = self.proof_subdir / 'proof.json'
-        dct: dict[str, list[int] | list[str] | bool | str | int | dict[int, str] | dict[int, list[dict[str, Any]]]] = {}
+        dct: dict[
+            str, list[int] | list[str] | bool | float | str | int | dict[int, str] | dict[int, list[dict[str, Any]]]
+        ] = {}
```
Comment (tothtamas28) [inline src/pyk/proof/reachability.py:450]:
```suggestion
        dct: dict[str, Any] = {}
```

File: src/pyk/proof/reachability.py [hunk 2]
```diff
@@ -53,6 +54,7 @@ class APRProof(Proof, KCFGExploration):
     logs: dict[int, tuple[LogEntry, ...]]
     circularity: bool
     failure_info: APRFailureInfo | None
+    execution_time: float
```
Comment (tothtamas28) [inline src/pyk/proof/reachability.py:57]: Let's make this private. Also, consider a shorter name:

```suggestion
    _exec_time: float
```

File: src/pyk/proof/reachability.py [hunk 3]
```diff
@@ -160,6 +164,12 @@ def prune(self, node_id: NodeIdLike, keep_nodes: Iterable[NodeIdLike] = ()) -> l
             self._bounded.discard(nid)
         return pruned_nodes
 
+    def update_execution_time(self, execution_time: float) -> None:
```
Comment (tothtamas28) [inline src/pyk/proof/reachability.py:167]:
```suggestion
    def add_exec_time(self, execution_time: float) -> None:
```

File: src/pyk/proof/reachability.py [hunk 4]
```diff
@@ -160,6 +164,12 @@ def prune(self, node_id: NodeIdLike, keep_nodes: Iterable[NodeIdLike] = ()) -> l
...
+        self.execution_time += execution_time
+
+    def reset_execution_time(self) -> None:
```
Comment (tothtamas28) [inline src/pyk/proof/reachability.py:170]:
```suggestion
    def reset_exec_time(self) -> None:
```

File: src/pyk/proof/reachability.py (hunk 2 above)
Comment (tothtamas28) [reply src/pyk/proof/reachability.py:57]: You'll then need a way to read the value, for that you can add a read-only property:

```python
def exec_time(self) -> float:
    return self._exec_time
```

**Comment by tothtamas28** [comment]
Eventually, such metadata should be provided by a generic runner of provers (but I understand that right now that's not an option).

File: src/pyk/proof/reachability.py (hunk 2 above)
Comment (lisandrasilva) [reply src/pyk/proof/reachability.py]: Makes sense. Thank you!

File: src/pyk/proof/reachability.py (hunk 4 above)
Comment (lisandrasilva) [reply src/pyk/proof/reachability.py]: Actually, I changed this function, it is now `set_exec_time(self, exec_time: float)` and updates the `self._exec_time` to the value of the received argument.

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**